Fix REPORT response from webdav (#4485)
owncloud/ocis: changelog/unreleased/fix-search-report.md

Bugfix: Fix search report

> There were multiple issue with REPORT search response from webdav. Also we want it to be consistent with PROPFIND endpoint.

Reviewer: ```suggestion
There were multiple issues with REPORT search responses from webdav. Also we want it to be consistent with PROPFIND responses.
```

*   the `remote.php` prefix was missing from the href (added even though not neccessary)

> *   the ids were formated wrong, they should look different for shares and spaces.

Reviewer: ```suggestion
*   the ids were formatted wrong, they should look different for shares and spaces.
```

*   the name of the resource was missing
*   the shareid was missing (for shares)
*   the prop `shareroot` (containing the name of the share root) was missing

> *   the  permissions prop was empty

Reviewer: ```suggestion
*   the permissions prop was empty
```
